app/models/intake.rb

```diff
@@ -519,20 +515,28 @@ def current_ticket_status
 
   private
 
-  def group_id_for_source
-    EitcZendeskInstance::ORGANIZATION_SOURCE_PARAMETERS.each do |key, value|
-      if source.downcase.starts_with?(key.to_s)
-        return value
-      end
-    end
-    nil
+  def partner_for_source
+    return nil unless source.present?
+
+    partner = SourceParameter.find_by(code: source.downcase)&.vita_partner
+    return nil unless partner.present?
+
+    [partner, "source_parameter", source]
   end
 
-  def group_id_for_state
-    EitcZendeskInstance::GROUP_ID_TO_STATE_LIST_MAPPING.each do |group_id, state_list|
-      return group_id if state_list.include? state_of_residence
-    end
+  def partner_for_state
+    state = State.find_by(abbreviation: state_of_residence&.upcase || state&.upcase)
```

Comment:
having two different cases (uppercase for state models, lowercase for intake records) I think bring some risk of bugs in the future. I think we should be consistent across all the models, but don't have an opinion on which one we use. Should we convert everything to uppercase, convert states to lowercase, or risk it?

Reply:
yes. i've removed the `STATE_OPTIONS` structure from `app/helpers/states.rb`, replacing with a dynamically generated list (without messing with the rest of the helpers). also, a backfill rake task `backfill:upcase_states` has been added to upcase all state field references on `Intake`.

Reply:
Would we need to run this backfill during deploy? Is it already set up to run automatically when we deploy this feature? Is there a risk if we run it manually after we deploy?